### PR TITLE
added no-redeclare rule

### LINKED ISSUE #1420: eslint: no-redeclare rule
We have eslint linting for our JS. Let's add a rule for [no-redeclare](https://eslint.org/docs/rules/) for consistency.

- [ ] Add the rule no-redeclare to .eslintrc.json 
- [ ] Run npm run lint:js and fix all offenders

### PR DESCRIPTION
**Pull request by redpfire**
This PR resolves #1420 

I've added the `no-redeclare` rule as stated in the issue and fixed all redeclarations.

Hope you merge.
Best regards.

### CODE REVIEW

File: openlibrary/plugins/openlibrary/js/lazy.js [hunk 1]
```diff
@@ -945,7 +945,7 @@ jQuery.fn.removeHighlight = function() {
 };
 
 //jTruncate
-(function($){$.fn.jTruncate=function(h){var i={length:300,minTrail:20,moreText:"more",lessText:"less",ellipsisText:"...",moreAni:"",lessAni:""};var h=$.extend(i,h);return this.each(function(){obj=$(this);var a=obj.html();if(a.length>h.length+h.minTrail){var b=a.indexOf(' ',h.length);if(b!=-1){var b=a.indexOf(' ',h.length);var c=a.substring(0,b);var d=a.substring(b,a.length-1);obj.html(c+'<span class="truncate_ellipsis">'+h.ellipsisText+'</span>'+'<span class="truncate_more">'+d+'</span>');obj.find('.truncate_more').css("display","none");obj.append('<div class="clearboth">'+'<a href="#" class="truncate_more_link">'+h.moreText+'</a>'+'</div>');var e=$('.truncate_more_link',obj);var f=$('.truncate_more',obj);var g=$('.truncate_ellipsis',obj);e.click(function(){if(e.text()==h.moreText){f.show(h.moreAni);e.text(h.lessText);g.css("display","none")}else{f.hide(h.lessAni);e.text(h.moreText);g.css("display","inline")}return false})}}})}})(jQuery);
```
Comment (jdlrobson) [inline openlibrary/plugins/openlibrary/js/lazy.js]: Please don't change the minified content here (don't fix any of the offenders in this file). A lot of this code is from vendors and we'll want to move it out. Changing it makes it harder for us to work out what versions they are. Instead, put this directive at the top of the file
```
 // eslint-disable
```

Comment (redpfire) [reply openlibrary/plugins/openlibrary/js/lazy.js]: Sure! I'll get right into it. One question. Should I set `no-redeclaration` to `warn` or just ignore errors in this file?

Comment (redpfire) [reply openlibrary/plugins/openlibrary/js/lazy.js]: Sure! I'll get right into it.

**Comment by jdlrobson** [comment]
Doesn't look like this got merged! Just approved!

**Comment by jdlrobson** [comment]
Opened https://github.com/internetarchive/openlibrary/pull/1543 which doesn't have the rebase conflicts.

**Comment by redpfire** [comment]
Got it :) Thanks for merging!

**Comment by mekarpeles** [comment]
Thank you @redpfire + @jdlrobson !

**Pull request closed without merging**